klaus/contrib/autodiscover.py

```diff
@@ -0,0 +1,194 @@
+import os
+import logging
+import threading
+import sys
+
+import dulwich
+
+from klaus.repo import FancyRepo
+from klaus import Klaus, make_app
+
+try:
+    from watchdog.observers import Observer
+    from watchdog.events import FileSystemEventHandler
+except ImportError:
+    sys.stderr.write("To automatically detect and reload repositories as they "
+                     "change, watchdog must be installed.\n")
+    sys.stderr.write("See https://pypi.python.org/pypi/watchdog/\n")
+    sys.exit(1)
+
+logger = logging.getLogger(__name__)
+
+
+def is_subdirectory(haystack, needle):
+    """
+    checks whether `needle` exists as a subdirectory of `haystack`
+    """
+    haystack = os.path.realpath(haystack)
+    needle = os.path.realpath(needle)
+    if not os.path.isdir(needle):
+        return False
+    return needle.startswith(haystack)
```

Comment:
You'll have to make sure both `haystack` and `needle` end with a `/` for the `.startswith` to work correctly. Counter example with the current implementation: `is_subdirectory("/foo", "/foo2")` is `True` if both paths exists, but should be False